feat: create Markdown primitive

kimak opened the pull request:
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [ ] Design validated

Commit:
feat: create Markdown primitive

Commit:
feat: create Markdown primitive

Commit:
fix italic & spacer

Commit:
update iOS screenshot

Commit:
update android screenshots

judastheo committed:
New design for ButtonIcon (#60)

* New design for ButtonIcon

* Update screenshots for components using ButtonIcon

---------

Co-authored-by: Théo Judas <theo@getluko.com>

kimak committed:
feat: theme adjust shadow values (#57)

* feat: theme adjust shadow values

* add shadows android screenschot

* fix contentContainerStyle Sandbox

* remove Box

* demo Shadow validation (to delete)

* sync shadow validation

* ShadowValidation iteration with expo

* update iOS screenshots

* revert App.tsx

* update android screenshot (Pixel5 - API33)

* remove shadowsValidation.doc.tsx

---------

Co-authored-by: Mickael Dumand <mickael@getluko.com>

Commit:
create markdown-link for CardTip

Commit:
fix conflict

kimak committed:
Merge branch 'dev' into md/feat/markdown

Commit:
update android CardTip screenshot

kimak commented on packages/streamline/src/primitives/markdown/markdown-link.tsx:
```diff
@@ -0,0 +1,55 @@
+    marginTop: 0,
+    marginBottom: 0,
+  },
+});
```
@judastheo I created a new MarkdownLink component and used it in CarTip
The goal is to avoid too much coupling with Markdown component. 

I wait for your approval on the dedicated commit before merging
https://github.com/GetLuko/streamline/pull/61/commits/e36bf6c99a9e71d28727fdb828e42f70f655e15b

kimak commented on packages/sandbox/e2e/android/screenshots/Markdown.png:
@ianrobertlambert waiting your approval on Android before merging

kimak commented on packages/sandbox/e2e/ios/screenshots/Markdown.png:
@ianrobertlambert waiting your approval on iOS before merging

kimak replied:
@ianrobertlambert I will merge it from a code point of view. Also because we already synced. But let me know if you have any feedback and I can iterate

judastheo reviewed (approved):
@kimak Good compromise I think, all clear!

ianrobertlambert commented:
Hey @kimak - As we discussed in sync, thanks for all of this! After taking a closer look, I had some small feedback (hopefully 🫠).

- [ ] Could we standardize the indentation to be 48px from the margin for visual alignment as you move down the screen?
<img width="435" alt="image" src="https://user-images.githubusercontent.com/34472045/236258239-e245f7eb-8dfb-49a6-ac7b-c04ca2c67158.png">

- [ ] It would be helpful to show how images with different aspect ratios, such as `16:9,` `1x1`, and `2x3`, can be scaled to demonstrate this to stakeholders.
![image](https://user-images.githubusercontent.com/34472045/236402620-b960a0a7-a2f9-4204-9928-3a3084090c96.png)

- [ ] Can we please add a `8px` border radius and `4px` of padding to the left and right of the inline code.
![image](https://user-images.githubusercontent.com/34472045/236403139-11436bec-a595-4ec4-9837-3fe4e4341e73.png)

- [ ] The type size for blockquotes seems slightly small; are we using `Title SM` here?

- [ ] On Android, the Bold and Italic Bold formatting looks like it's using the wrong typeface. Could it be a rendering issue for me?

- [ ] The numbers on the ordered list seem a bit small; are we using the `Body` typeface here?

- [ ] It would be easier to provide accurate feedback if we can see the examples wrap. I've made some quick adjustments to the markdown sandbox page, and it could help to see how the markdown is used in context between elements. WDYT? https://www.figma.com/file/tOJLNigXQXJqCSKgJFPZn5/App-Streamline-Lukomponents?node-id=5448%3A15280&t=XTDSSy8UfzS7PCu9-1

ianrobertlambert commented:
Sorry, I think this feedback on Markdown was lost because it was merged. Should I create a new issue? cc @judastheo @kimak @hcourthias

kimak commented:
thanks for pointing it @ianrobertlambert I missed it. I will take your feedback today

kimak commented:
@ianrobertlambert here is my update:

> - [x] Could we standardize the indentation to be 48px from the margin for visual alignment as you move down the screen?

> - [x] It would be helpful to show how images with different aspect ratios, such as `16:9,` `1x1`, and `2x3`, can be scaled to demonstrate this to stakeholders.

I added it at the end so you should scroll with expo app to see them.

> - [ ] Can we please add a `8px` border radius and `4px` of padding to the left and right of the inline code.
![image](https://user-images.githubusercontent.com/34472045/236403139-11436bec-a595-4ec4-9837-3fe4e4341e73.png)

It's not possible with the actual implementation of the lib because the text should wrap with other texts.
This markdown was not in Figma initially, should we keep or delete?

> - [x] The type size for blockquotes seems slightly small; are we using `Title SM` here?

Good catch! It's because blockquote should be wrap as title: `> # Title quote example` 
I updated the expo example. 

> - [ ] On Android, the Bold and Italic Bold formatting looks like it's using the wrong typeface. Could it be a rendering issue for me?

Seems good on my side. Could you check the Luko app version with Markdown? it's the same font, but module fonts can differ on expo.

> - [x] The numbers on the ordered list seem a bit small; are we using the `Body` typeface here?

Good catch. Fixed

> - [x] It would be easier to provide accurate feedback if we can see the examples wrap. I've made some quick adjustments to the markdown sandbox page, and it could help to see how the markdown is used in context between elements. WDYT? https://www.figma.com/file/tOJLNigXQXJqCSKgJFPZn5/App-Streamline-Lukomponents?node-id=5448%3A15280&t=XTDSSy8UfzS7


Also added when you scroll, example here: 

![Screenshot 2023-06-13 at 18 24 15](https://github.com/GetLuko/streamline/assets/1573707/7f2db29f-b001-4018-a6a3-be83885fbf94)
![Screenshot 2023-06-13 at 18 24 28](https://github.com/GetLuko/streamline/assets/1573707/1a331dff-355d-4ef2-96f0-a518fac3c462)
![Screenshot 2023-06-13 at 18 24 42](https://github.com/GetLuko/streamline/assets/1573707/ef2a0f83-ece2-48ca-8ae4-a4a85f71acb3)
